test/api/tenants.test.ts

```diff
@@ -0,0 +1,175 @@
+      roles: {
+        response: [
+          "'bap_plugins_visibility' created.",
+          "'bap_new_tenant_name_privileged_user_role' created.",
+          "'bap_new_tenant_name_pseudonymize_role' created.",
+          "'bap_new_tenant_name_user_role' created.",
+          "'bap_new_tenant_name_mordred_role' created.",
+        ],
+      },
+      backendRoles: {
+        response: [
+          "'bap_new_tenant_name_user_role' created.",
+          "'bap_plugins_visibility' created.",
+          "'bap_new_tenant_name_pseudonymize_role' created.",
+          "'bap_new_tenant_name_privileged_user_role' created.",
+        ],
+      },
+      dashboards: expect.anything(),
+    });
+
+    const roles = response.body.data.roles.response;
+    expect(roles).not.toContain(
+      "'bap_new_tenant_name_anonymous_access_role' created"
+    );
+  });
+
+  it('Creates anonymous role', async () => {
+    // Delete 'bap_plugins_visibility' role from last test
+    await osdTestServer.request
+      .delete(root, '/api/v1/configuration/roles/bap_plugins_visibility')
+      .set('Authorization', credentials);
+
+    const response = await osdTestServer.request
+      .post(root, '/api/create_tenant')
+      .set('Authorization', credentials)
+      .send({
+        anonymous: true,
+        tenant: 'anonymous_tenant_name',
+      });
+
+    expect(response.status).toEqual(200);
+
+    expect(response.body.data).toMatchObject({
+      tenant: "'anonymous_tenant_name' created.",
+      roles: {
+        response: [
+          "'bap_plugins_visibility' created.",
+          "'bap_anonymous_tenant_name_anonymous_access_role' created.",
+          "'bap_anonymous_tenant_name_privileged_user_role' created.",
+          "'bap_anonymous_tenant_name_pseudonymize_role' created.",
+          "'bap_anonymous_tenant_name_user_role' created.",
+          "'bap_anonymous_tenant_name_mordred_role' created.",
+        ],
+      },
+      backendRoles: {
+        response: [
+          "'bap_anonymous_tenant_name_privileged_user_role' created.",
+          "'bap_anonymous_tenant_name_mordred_role' created.",
+          "'bap_anonymous_tenant_name_anonymous_access_role' created.",
+          "'bap_anonymous_tenant_name_user_role' created.",
+          "'bap_anonymous_tenant_name_pseudonymize_role' created.",
+        ],
+      },
+      dashboards: expect.anything(),
+    });
+  });
+
+  it('Fails if tenant already exists', async () => {
+    const response = await osdTestServer.request
+      .post(root, '/api/create_tenant')
+      .set('Authorization', credentials)
+      .send({
+        force: false,
+        tenant: 'new_tenant_name',
+      });
+
+    expect(response.status).toEqual(400);
+  });
+
+  it('Overwrites roles and tenant', async () => {
+    const response = await osdTestServer.request
+      .post(root, '/api/create_tenant')
+      .set('Authorization', credentials)
+      .send({
+        force: true,
+        tenant: 'new_tenant_name',
+      });
+
```

Comment:
I miss a test that cleans up everything when the roles already exist.

Reply:
I added a test that checks that no tenant or roles are created if one of the roles already exists.